### PR TITLE
Adding fresh playwright vst

### PR DESCRIPTION
**Pull request by bfuzzy1**

### CODE REVIEW

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/playwright.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            # Github test actions
+            setup_github_test_action(page)
+            time.sleep(10)
+            edit_repository(page)
+            time.sleep(10)
+            changepassword(page)
+            time.sleep(10)
+            sys.exit(100)
+        else:
+        # Login was not successful
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/playwright.py:118]: indentation looks off

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"os"
+
+	Endpoint "github.com/preludeorg/test/endpoint"
+)
+
+//go:embed playwright.py
+var playwright []byte
+
+func installed() {
+		task := fmt.Sprintf("playwright --version")
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go:20]: is it just me or is the indentation all over the place in this PR?

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		stdout := Endpoint.Run(task)
+		println(stdout)
+	}
+
+func test() {
+ 	os.Setenv("GITHUB_REPO_NAME", "")
+ 	os.Setenv("GITHUB_USERNAME", "")
+ 	os.Setenv("GITHUB_PASSWORD", "")
+
+ 	Endpoint.Write(os.TempDir()+"\\playwright.py", playwright)
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go:30]: make ``os.TempDir()+"\\playwright.py", playwright`` a variable?

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+	Endpoint "github.com/preludeorg/test/endpoint"
+)
+
+//go:embed playwright.py
+var playwright []byte
+
+func installed() {
+		task := fmt.Sprintf("playwright --version")
+		stdout := Endpoint.Run(task)
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go:21]: should exit with status of 104 if not relevant.

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	Endpoint "github.com/preludeorg/test/endpoint"
+)
+
+//go:embed playwright.py
+var playwright []byte
+
+func installed() {
+	task := fmt.Sprintf("playwright --version")
+	stdout := Endpoint.Run(task)
+	println(stdout)
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go:22]: psst; should exit with 104 if playwright is not installed

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+		if err != nil {
+			if exitError, ok := err.(*exec.ExitError); ok {
+				return exitError.ExitCode(), string(stdout), string(exitError.Stderr)
+			}
+		}
+		return 0, string(stdout), ""
+	}
+}
+
+func runPython(command string) (int, string, string) {
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go:46]: we need a single 'run' function in Endpoint that is capable of running any system command for any installed application.

**Comment by bfuzzy1** [comment]
Added fresh playwright script. Single function that will run through a few different actions on a single login.

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+	exitCode := Endpoint.Remove(scriptPath)
+	os.Exit(exitCode)
+}
+
+func main() {
+	args := os.Args[1:]
+	if len(args) > 0 {
+		clean()
+	} else {
+		installed()
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go:98]: main function should not be touched. put installed in the test.

**Review by privateducky — COMMENTED** [review]
ok; this looks clean. 
let me know how testing goes.

File: tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go [hunk 8]
```diff
@@ -0,0 +1,74 @@
+/*
+FILENAME: 61da9575-5c7e-471a-a3d7-27db8c7eea38.go
```
Comment (privateducky) [inline tests/61da9575-5c7e-471a-a3d7-27db8c7eea38/61da9575-5c7e-471a-a3d7-27db8c7eea38.go:2]: change filename to ID

**Review by privateducky — CHANGES_REQUESTED** [review]
1. Add a README.
2. Is this tested?

**Comment by bfuzzy1** [comment]
> * Add a README.
> * Is this tested?

Playwright has no support for ARM install for Windows.

https://pypi.org/project/playwright/#files

Screenshot 2023-01-23 at 10 51 59 AM

**Comment by VVX7** [comment]
confirming this works correctly. tested on windows 11 arm64 with python amd64.

there's an edge case where if the target github account doesn't contain at least one repo the script will time out. recommend adding this requirement in the readme @bfuzzy1

**Review by privateducky — CHANGES_REQUESTED** [review]
please update to Endpoint.Start and Endpoint.Stop logic.

**Comment by bfuzzy1** [comment]
Made a change to the test function.

This has passed testing.

**Pull request merged**